lib/engine/step/train.rb

```diff
@@ -61,6 +61,9 @@ def buy_train_action(action, entity = nil)
 
         source = @depot.discarded.include?(train) ? 'The Discard' : train.owner.name
 
+        @game.phase.buying_train!(entity, train)
```

Comment:
i wonder how if this will break games

Reply:
i'm pretty sure this change is going to break hundreds of games

Reply:
I do not think it will break any games but I have only tested with rake tests, and it passes them.

The implementation before did like this:
- Oh, I would like to buy this train from the depot
- Ok, before doing that let's change phase if that train triggers a phase change
- And now do give us the cheapest train if we add new trains for the next phase

For all other game titles implemented the new trains are not cheaper than the train triggering phase change. But for 18MS 6 train cost $550, and the newly available 2D cost $500. This means that old implementation stops you from buying the 6 train....

But is it possible to try this change with all games to see if it does break any game at all?

Reply:
i think the issue will be that the buying_train triggers train rusting allowing people to buy the next train even though it becomes traight tight

Reply:
yes, i can try this change before we deploy it

Reply:
I do not see that this should cause any problems. The check if you are allowed to buy any trains is done with the phase that are active before you press the button. Besides it is a bug in the current implementation that I am trying to fix.

But trying it out on the existing games should clarify if there is a problem. At least the spec test games does not report any problem.